docs/testplan/PMON-Chassis-Enhancements-test-plan.md

```diff
@@ -0,0 +1,672 @@
+ * Verify output of each line card   shows 'up' for operational state admin state for itself and  supervisor for DUT
+
+### 1.2 Configure chassis-modules 
+
+#### Steps
+
+ * Shutdown line card use Run command ```sudo config chassis-modules shutdown <card>```
+ * Startup line card use Run command ``` sudo config chassis-modules startup <card> ```
+
+#### Repeat steps on
+ * Supervisor
+ * Line Cards
+
+#### Pass/Fail Criteria
+ * Verify ``` show chassis-modules status ``` report line-card down after shutdown on supervisor
+
+    ```
+           sudo config chassis-modules shutdown LINE-CARD1
+           admin@sonic:~$ show chassis-modules status LINE-CARD1 
+                       Name        Description     	Slot    Oper-Status    Admin-Status
+                    -------------  -----------------  ------  -------------  --------------
+                    LINE-CARD1  imm36-400g-qsfpdd       1       down            down
+    ```
+     
+
+ * Verify  ``` show chassis-modules status ``` report line-card up after startup on supervisor
+     for example:
+    
+      ```
+         sudo config chassis-modules startup LINE-CARD1
+         admin@sonic:~$ show chassis-modules status LINE-CARD1 
+                   Name        Description     	Slot    Oper-Status    Admin-Status
+                -------------  -----------------  ------  -------------  --------------
+                LINE-CARD1  imm36-400g-qsfpdd       1       Online           up
+      ```
+ *  ``` show chassis-modules status ``` report card status unchanged after shutdown on line card
+ 
+
+### 1.3 Check platform chassis midplane status 
+
+#### Steps
+ * Use command ` show chassis-modules midplane-status ` to get status of midplane in modular chassis
+
+#### Verify in
+ * Supervisor
+ * Line Card
+
+#### Sample Output
+     ```
+       on Supervisor:
+         show chassis-modules midplane-status
+              Name    IP-Address    Reachability
+        ----------  ------------  --------------
+        LINE-CARD0      10.0.0.1            True
+        LINE-CARD1      10.0.0.2           False
+        LINE-CARD2      10.0.0.3            True
+        LINE-CARD3      10.0.0.4            True
+    ```  
+     ```
+          on line card: 
+                     Name    IP-Address    Reachability
+                -----------  ------------  --------------
+                SUPERVISOR0     10.0.0.16            True
+
+     ```
+     
+
+#### Pass/Fail Criteria
+ * Verify output on supervisor lists all the line cards, midplane ip addresses and reachability status as expected  
+ * Verify on each Line Card supervisor reachability is correct and supervisor ip address is correct
+
+### 1.4 Check thermal sensor output in chassis state db 
+#### Steps
+ * Run command “redis-dump  -p 6380 -d 13 -y -k "*TEMP*" "on Supervisor
+ * Run command “redis-dump -d 6 -y -k "*TEMP*" on line card
```

Comment:
Suggest surrounding commands with single backticks rather than quotes to produce inline code.